### PR TITLE
SDK-2264 share v2 net-retrieve-receipt

### PR DESCRIPTION
**Pull request by mehmet-yoti**
SDK-2264:Added GetShareReceipt and CryptoEngine methods,added models, unit tests
SDK-2264:Added example digitalidentity project, controllers

### CODE REVIEW

File: src/Examples/DigitalIdentity/DigitalIdentity/docker-compose.dcproj [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (irotech) [inline src/Examples/DigitalIdentity/DigitalIdentity/docker-compose.dcproj:1]: what is this for?

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs [hunk 2]
```diff
@@ -156,8 +162,212 @@ internal static async Task<GetQrCodeResult> GetQrCode(HttpClient httpClient, Uri
...
+                    },
+                    Error = receiptResponse.Error
+                };
+
+                return sharedReceiptResponse;
+            }
+            catch  (Exception ex)
+            {
+                throw new Exception($"An unexpected error occurred: {ex.Message}");
+
```
Comment (irotech) [inline src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs:298]: unnecessary empty line

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs [hunk 3]
```diff
@@ -156,8 +162,212 @@ internal static async Task<GetQrCodeResult> GetQrCode(HttpClient httpClient, Uri
...
+                var encryptedItemKeyResponse = await GetReceiptItemKey(httpClient, itemKeyId, clientSdkId, apiUrl, key);
+
+                var wrappedKey = Encoding.ASCII.GetBytes(receiptResponse.WrappedKey);
+                var receiptContentKey = CryptoEngine.UnwrapReceiptKey(wrappedKey, encryptedItemKeyResponse.Value, encryptedItemKeyResponse.Iv, key);
+ 
+                var (attrData, aextra, decryptAttrDataError) = DecryptReceiptContent(receiptResponse.Content, receiptContentKey);
+                if (decryptAttrDataError != null)
+                {
+                    throw new Exception($"An unexpected error occurred: {decryptAttrDataError.Message}");
+
```
Comment (irotech) [inline src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs:252]: unnecessary empty line

**Comment by irotech** [comment]
Great job in re-arranging the classes and the commits. It's now much easier to go through. I did the first pass, first thing tomo morning I'll do another one.

File: src/Examples/DigitalIdentity/DigitalIdentity/Controllers/HomeController.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                    .WithGender()
+                    .WithDocumentDetails()
+                    .WithDocumentImages()           
+                    .Build();
+
+                var sessionReq = new ShareSessionRequestBuilder().WithPolicy(policy)
+                    .WithRedirectUri("https:/www.yoti.com").WithSubject(new
+                    {
+                        subject_id = "some_subject_id_string"
+                    }).Build();
```
Comment (irotech) [inline src/Examples/DigitalIdentity/DigitalIdentity/Controllers/HomeController.cs:55]: Fluent API reads better as one line per assignment, something like

```
var sessionReq = new ShareSessionRequestBuilder()
    .WithPolicy(policy)
    .WithRedirectUri("https:/www.yoti.com")
    .WithSubject(new { subject_id = "some_subject_id_string" })
    .Build();
```

or 
```
var sessionReq = new ShareSessionRequestBuilder()
    .WithPolicy(policy)
    .WithRedirectUri("https:/www.yoti.com")
    .WithSubject(
        new { 
            subject_id = "some_subject_id_string" 
        }
    )
    .Build();
```

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs [hunk 5]
```diff
@@ -156,8 +162,212 @@ internal static async Task<GetQrCodeResult> GetQrCode(HttpClient httpClient, Uri
...
+                        UserProfile = userProfile,
+                        ExtraData = userExtraData
+                    },
+                    ApplicationContent = new ApplicationContent
+                    {
+                        ApplicationProfile = applicationProfile,
+                        ExtraData = appExtraData
+                    },
+                    Error = receiptResponse.Error
+                };
```
Comment (irotech) [inline src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs:291]: I don't know how it works in .NET but few properties are optional in the receipt (`rememberMeId`, `parentRememberMeId` for instance) make sure to set these values only if not null

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs [hunk 6]
```diff
@@ -156,8 +162,212 @@ internal static async Task<GetQrCodeResult> GetQrCode(HttpClient httpClient, Uri
...
+                        receiptResponse.WrappedKey,
+                        receiptResponse.Content.ExtraData,
+                        key);
+
+                var sharedReceiptResponse = new SharedReceiptResponse
+                {
+                    ID = receiptResponse.ID,
+                    SessionID = receiptResponse.SessionID,
+                    RememberMeID = receiptResponse.RememberMeID,
+                    ParentRememberMeID = receiptResponse.ParentRememberMeID,
```
Comment (irotech) [inline src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs:278]: Are you Base64 encoding these 2 from the HTTP response?

**Review by irotech — APPROVED** [review]

**Review by saurabh-yoti — APPROVED** [review]

**Review by saurabh-yoti — APPROVED** [review]

**Pull request merged**